test/monitoring_console/monitoring_console_test.go

```diff
@@ -0,0 +1,513 @@
+					if strings.Contains(peer, podName) {
+						found[podName] = true
+						break
+					}
+				}
+			}
+			allSearchHeadsConfigured := true
+			for _, key := range found {
+				if !key {
+					allSearchHeadsConfigured = false
+					break
+				}
+			}
+			Expect(allSearchHeadsConfigured).To(Equal(true))
+
+			// Scale Search Head Cluster
+			testenvInstance.Log.Info("Scaling search head cluster")
+			_, err = exec.Command("kubectl", "scale", "shc", "-n", testenvInstance.GetName(), deployment.GetName(), "--replicas=4").Output()
+			Expect(err).To(Succeed(), "Failed to scale search head cluster")
+
+			// Ensure search head cluster go to ScalingUp phase
+			Eventually(func() splcommon.Phase {
+				err := deployment.GetInstance(deployment.GetName(), shc)
+				if err != nil {
+					return splcommon.PhaseError
+				}
+				testenvInstance.Log.Info("Waiting for search head cluster instance status to be Scaling Up", "instance", shc.ObjectMeta.Name, "Phase", shc.Status.Phase)
+				dumpGetPods(testenvInstance.GetName())
+				return shc.Status.Phase
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(splcommon.PhaseScalingUp))
+
+			// Ensure search head cluster go to Ready phase
+			Eventually(func() splcommon.Phase {
+				err := deployment.GetInstance(deployment.GetName(), shc)
+				if err != nil {
+					return splcommon.PhaseError
+				}
+				testenvInstance.Log.Info("Waiting for search head cluster instance status to be READY", "instance", shc.ObjectMeta.Name, "Phase", shc.Status.Phase)
+				dumpGetPods(testenvInstance.GetName())
+				return shc.Status.Phase
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// In a steady state, we should stay in Ready and not flip-flop around
+			Consistently(func() splcommon.Phase {
+				_ = deployment.GetInstance(deployment.GetName(), shc)
+				return shc.Status.Phase
+			}, ConsistentDuration, ConsistentPollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// Check New MC Pod comes up and old one is terminated
+			Eventually(func() int {
+				mcPodCount := testenv.GetMCPodCount(testenvInstance.GetName())
+				return mcPodCount
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(1))
+
+			// Monitoring Console Pod is in Ready State
+			Eventually(func() bool {
+				check := testenv.CheckMCPodReady(testenvInstance.GetName())
+				return check
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(true))
+
+			// Check Monitoring console is configured with all search head instances in namespace
+			peerList = testenv.GetConfiguredPeers(testenvInstance.GetName())
+			found = make(map[string]bool)
+			testenvInstance.Log.Info("Peer List", "instance", peerList)
+			for i := 0; i < 4; i++ {
+				podName := "splunk-" + deployment.GetName() + "-search-head-" + strconv.Itoa(i)
+				found[podName] = false
+				for _, peer := range peerList {
+					testenvInstance.Log.Info("Check Peer matches serach head pod", "Search Head Pod", podName, "Peer in peer list", peer)
```

Comment:
This log should go inside the if block?

Reply:
It is very useful to have it outside in case of a failure make debugging much easier when looking at logs on Circle CI. However if log bloat is an issue I can move it inside the if block